src/modules/logger/logger.cpp

```diff
@@ -268,7 +266,11 @@ Logger *Logger::instantiate(int argc, char *argv[])
 			break;
 
 		case 'e':
-			log_on_start = true;
+			log_mode = Logger::LogMode::boot_until_disarm;
```

Comment:
This will not be correct if `-f -e` is specified. This is why the logic is as it is now.

Reply:
@bkueng, thanks again. but I think you wrong here. this is exactly what I wanted to fix here, the confusion.
as default log_on_start = false; log_until_shutdown = false;

-e set log_on_start = true;
-f set log_on_start = true; log_until_shutdown = true;

later:
	if (log_on_start) {
		if (log_until_shutdown) {	log_mode = Logger::LogMode::boot_until_shutdown;
		} else { log_mode = Logger::LogMode::boot_until_disarm;}}

any chance you ment if `-f -e` is specified? that can't be

Reply:
Yes I meant if both are specified, which is valid and correct (`-f` implies `-e`).

Reply:
@bkueng - to be clear, there is no fix required here? its all good, and more clear now, right?

Reply:
No, it's incorrect now. Please revert back to how it was.

Reply:
@bkueng It took me some time to understand you. I see now what you meant. My thought is that its either using -e or -f or -x. and not some kind of combinations.
on the previous algorithm using '-f' or '-f -e' or '-e -f' did the exact some thing.
On my "way" is only one of them.
is there way to enforce it? and yes, it breaks something on current confusing use.
And I don't talk about how -x confuse the current logic....

Reply:
From a CLI perspective this is not confusing, it's clearly documented that -f imples -e.
Only in code there's a bit more logic to handle this. I don't see a problem with this.
Changing the CLI is an option, but then you must return an error if both are provided - the end result won't really be simpler.

Reply:
@bkueng - I can't say that I am 100% agree, but I have changed to logic according to your recommendation on [this commit](https://github.com/BazookaJoe1900/Firmware/commit/dbac8513c58d419854ab633257ea854e763140f9)

Reply:
After some rebasing issue... Ready to re-review